lib/CCreatureHandler.cpp

```diff
@@ -392,6 +392,20 @@ void CCreature::fillWarMachine()
 	warMachine = ArtifactID::NONE; //this creature is not artifact
 }
 
+void CCreature::updateOppositeBonuses()
+{
+	auto & bonusList = getExportedBonusList();
+	for(auto & bonus : bonusList)
+	{
+		if(bonus->effectRange == Bonus::ONLY_ENEMY_ARMY //Opposite Side bonuses should not be exported from CREATURE node.
+			|| (bonus->propagator && bonus->propagator->getPropagatorType() == CBonusSystemNode::BATTLE))

```

Comment:
Could you clarify case when we have BATTLE propagator and not ONLY_ENEMY_ARMY range? And what will be in this case?

Reply:
If you have a look at  config and mods, other effectRange values don't need specific propagator.
In other hand, BATTLE propagator is used only for Opposite Side Bonuses.
Although, it will be work with another limiters, not only with OppositeSideLimiter, but it will be extra overhead.
Thus, BATTLE with not ONLY_ENEMY_ARMY range is nonsense and it will be overwritten here.
In my instruction I suggest modders to use ONLY_ENEMY_ARMY and don't touch BATTLE propagator.

Reply:
non-leaving creature reducing all living creatures attack on the battlefield. Also how artifacts blocking magic on the battlefield work? Orb of inhibition if I am not mistaken? Also I suggest to at least print an error if we do override some propagator.

Reply:
Obviously, they don't need to use ONLY_ENEMY_ARMY effect range or Opposite limiter and it's not current discussion theme.

Reply:
You said BATTLE propagator is used only for Opposite Side Bonuses. I ask why cannot we use BATTLE propagator with creature type limiter or other bonus limiter?

Reply:
Take into account that you use ONLY_ENEMY_ARMY OR BATTLE propagator, not AND. So any BATTLE propagator has to use OppositeSideLimiter, am I correct?

Reply:
Particular part of code, which you are reviewing is used to update creature data from old saves. So, it's written considering current mods data. I know  only one attempt to use this propagator in 3thd party mod and that attempt looks wrong, Thus, it's possible to say, that BATTLE_WIDE prop. is not used for creatures now.
Propagator checking here is for additional insurance.

P. S. Creature node has no owner - that's why it's not suitable to be holder of bonus with particular 'OppositeSide' limiter